app/store.go

```diff
@@ -347,14 +347,14 @@ func (s *StoreApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBegi
 // Returns a list of all validator changes made in this block
 // TODO: investigate response tags as of 0.11 abci
 func (s *StoreApp) EndBlock(_ abci.RequestEndBlock) (res abci.ResponseEndBlock) {
-	res.ValidatorUpdates = s.pending
+	res.ValidatorUpdates = weave.ValidatorUpdatesToABCI(s.pending)
 	s.pending = nil
 	return
 }
 
 // AddValChange is meant to be called by apps on DeliverTx
 // results, this is added to the cache for the endblock changeset
-func (s *StoreApp) AddValChange(diffs []abci.ValidatorUpdate) {
+func (s *StoreApp) AddValChange(diffs []weave.ValidatorUpdate) {
```

Comment:
Note that this is the same basic logic flow as in x/validators for applying a diff.
I will look more below much maybe `CombineUpdates(cur, diff []ValidatorUpdate)` would make sense in weave root.

Reply:
Yeah I actually looked at it to make sure we mimic it to avoid any inconsistencies with updates and bookkeeping. I think we should do it in one place to avoid maintaining two.